### ISSUE TITLE
tflint incorrectly warns on interpolation-only map attributes (terraform_deprecated_interpolation)

### ISSUE THREAD

**Issue by willhughes-au** [terraform-linters/tflint#1462]
## Introduction

With a recent change, tflint now returns a Warning when I have an interpolation-only string attribute in a map. 

Applying the suggestion results in invalid HCL that Terraform rejects. 

## Expected Behavior

It should not suggest/warn on interpolation-only string attributes in a map. 

## Actual behavior

Running tflint over the example playbook results in this warning: 

`Warning: Interpolation-only expressions are deprecated in Terraform v0.12.14 (terraform_deprecated_interpolation)` 

## Step to Reproduce

Run this example against tflint 0.39.1: 

```hcl
locals {
        foo = "bar"
}

output "map_a" { 
    value = {
        "${local.foo}" : "baz"
    }
}
```

## Additional Context

This is the (invalid) suggestion: 

```hcl
locals {
        foo = "bar"
}

output "map_b" { 
    value = {
        local.foo : "baz"
    }
}
```

Running this results in an error from Terraform: 

```
$ tf plan
╷
│ Error: Ambiguous attribute key
│ 
│   on test.tf line 12, in output "map_b":
│   12:         local.foo : "baz"
│ 
│ If this expression is intended to be a reference, wrap it in parentheses. If it's instead intended as a literal name containing periods, wrap it in quotes to create a string literal.
```

```console
$ tflint -v
TFLint version 0.39.1
$ terraform -v
Terraform v1.2.6
```

**Comment by willhughes-au** [comment]
I think this may be related to https://github.com/terraform-linters/tflint/issues/1257

**Comment by bendrucker** [comment]
What suggestion? The full output is as follows:

```
Warning: Interpolation-only expressions are deprecated in Terraform v0.12.14 (terraform_deprecated_interpolation)

  on main.tf line 7:
   7:         "${local.foo}" : "baz"

Reference: https://github.com/terraform-linters/tflint/blob/v0.39.1/docs/rules/terraform_deprecated_interpolation.md
```

This is indeed an unnecessary interpolation:

https://www.terraform.io/language/expressions/types#maps-objects

> You can use a non-literal string expression as a key by wrapping it in parentheses

Terraform gives you this hint right in the error message.

A correctly formatted key expression (parentheses, not interpolation) will not raise warnings in TFLint, both on older versions, and the latest.
